### PR TITLE
Catch IOException in Get-FileHash

### LINKED ISSUE #11931: Get-FileHash Terminates at Corrupt File
When running a script to obtain the hashes of all files in a given directory, PS 7.0.0-rc.3 will terminate the command when it comes across a corrupt file.  In PS 5.1, the same command will throw an error but continue generating hashes of additional files.  The script I;m using to test is 
```powershell
get-childitem h:\ -Recurse -Force -File | Get-FileHash
```
In either version of Powershell, I get the error below.  The only difference is that the script continues processing in 5.1 and terminates in 7.  

> Get-FileHash: The file or directory is corrupted and unreadable. : 'H:\Intel\gp\profile.dat'

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #11931
Catch all IO exceptions in Get-FileHash cmdlet to do not terminate pipeline.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs [hunk 1]
```diff
@@ -156,6 +156,16 @@ protected override void ProcessRecord()
                         path);
                     WriteError(errorRecord);
                 }
+                catch (IOException ioException)
+                {
+                    ThrowTerminatingError(
```
Comment (mklement0) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: This should be WriteError() too - as in the original WinPS implementation, any exception for a given pipeline input should be a non-terminating error.

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: Yes, It is was copy-past without fix.

**Review by daxian-dbw — COMMENTED** [review]
@iSazonov You can use the special `.sys` file on the system drive of a windows OS for the test, for example, `pagefile.sys` or `swapfile.sys`.

**Comment by iSazonov** [comment]
@daxian-dbw I added new test.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs [hunk 2]
```diff
@@ -156,6 +156,15 @@ protected override void ProcessRecord()
                         path);
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '10.79.1.228', but got 127.0.0.1.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Get-FileHash.Default result tests.Should write non-terminating error if a file is locked
```powershell
Expected strings to be the same, but they were different.
Expected length: 62
Actual length:   61
Strings differ at index 4.
Expected: 'FileReadEr...'
But was:  'FileNotFou...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-FileHash.Tests.ps1: line 35
35:             $errorVariable.FullyQualifiedErrorId | Should -BeExactly "FileReadError,Microsoft.PowerShell.Commands.GetFileHashCommand"
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '10.79.1.8', but got 127.0.0.1.
at <ScriptBlock>, /Users/runner/runners/2.165.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Get-FileHash.Default result tests.Should write non-terminating error if a file is locked
```powershell
Expected strings to be the same, but they were different.
Expected length: 62
Actual length:   61
Strings differ at index 4.
Expected: 'FileReadEr...'
But was:  'FileNotFou...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-FileHash.Tests.ps1: line 35
35:             $errorVariable.FullyQualifiedErrorId | Should -BeExactly "FileReadError,Microsoft.PowerShell.Commands.GetFileHashCommand"
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Get-FileHash.Default result tests.Should write non-terminating error if a file is locked
```powershell
Expected strings to be the same, but they were different.
Expected length: 62
Actual length:   61
Strings differ at index 4.
Expected: 'FileReadEr...'
But was:  'FileNotFou...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-FileHash.Tests.ps1: line 35
35:             $errorVariable.FullyQualifiedErrorId | Should -BeExactly "FileReadError,Microsoft.PowerShell.Commands.GetFileHashCommand"
```

**Comment by iSazonov** [comment]
Oh, It turns out CI Windows VM redirect pagefile.sys so I have to use WMI to find it in the new test.

**Comment by daxian-dbw** [comment]
Didn't know you can use cim cmdlet for that, very cool :)
LGTM.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**